reya-rust-sdk/src/http_provider.rs

```
@@ -298,25 +298,40 @@ impl HttpProvider {
 
         if receipt.inner.is_success() {
             debug!("BatchExecute receipt:{:?}", receipt);
-            let logs_result = self.get_transaction_receipt(receipt.transaction_hash).await;
-            match logs_result {
-                Some(logs) => {
-                    let mut i = 0;
-                    for log in logs {
-                        let log_data = log.data();
-                        //
-                        info!("received log, index={}, len:{}", i, log_data.data.len());
-                        // todo check the log_data and make a correct discision if it is successful or not
-                        set_batch_order_state(batch_orders, i, true);
-
-                        i += 1;
-                    }
-                }
-                None => {}
+        }
+
+        // todo: p2: consider returning a new list of batch orders with execution states to the client
+        // alongside the receipt instead of manipulating the list in memory
+        let logs = receipt.inner.logs();
+        self.parse_execute_batch_events(batch_orders, logs);
+
+        eyre::Ok(receipt)
+    }
+
+    fn parse_execute_batch_events(
+        &self,
+        batch_orders: &mut Vec<data_types::BatchOrder>,
+        logs: &[Log],
+    ) {
+        // todo: p2: ensure length of logs is the same as length of batch orders
```

indeed, what I expect here is that the length of the batch_orders is equal to the length of the logs. When this is not the case we should return an error result and the caller will need to react on that.

---

correct, lemme add that in as well